mountpoint-s3/src/autoconfigure.rs

```diff
@@ -0,0 +1,48 @@
+use anyhow::{anyhow, Context};
+
+use mountpoint_s3_client::instance_info::InstanceInfo;
+
+/// Determine the maximum network throughput for the current instance using IMDS. Returns an error
+/// if the instance type either cannot be retrieved using the IMDS client or does not have a known
+/// network throughput.
+pub fn network_throughput(instance_info: &InstanceInfo) -> anyhow::Result<f64> {
+    let instance_type = instance_info
+        .instance_type()
+        .map_err(|e| anyhow!("failed to get instance type: {e}"))?;
+    let throughput = get_maximum_network_throughput(instance_type).context("failed to get network throughput")?;
+    Ok(throughput)
+}
+
+fn get_maximum_network_throughput(ec2_instance_type: &str) -> anyhow::Result<f64> {
+    const INSTANCE_THROUGHPUT: &str = "instance_throughput";
```

Comment:
Not for this PR, but should we consider using the recommended configuration from https://github.com/awslabs/aws-c-s3/pull/367?

Reply:
Right now that API only supports a few hard-coded instance types: https://github.com/awslabs/aws-c-s3/blob/5fe1c3b116d4b2ac69304af2d2d3031e4b6d6d14/source/s3_platform_info.c#L328-L335

I think probably we should consult it first, and then fall back to our version if we don't get a match from it.

Reply:
I was hoping we could remove our list. Not really, unfortunately.